### PR TITLE
Cleanup TX db storage format

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/35439

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/solana_transaction.h [hunk 1]
```diff
@@ -51,21 +51,20 @@ class SolanaTransaction {
...
-  explicit SolanaTransaction(SolanaMessage&& message);
+  explicit SolanaTransaction(SolanaMessage message);
 
   SolanaTransaction(
       mojom::SolanaMessageVersion version,
       const std::string& recent_blockhash,
       uint64_t last_valid_block_height,
       const std::string& fee_payer,
       const SolanaMessageHeader& message_header,
-      std::vector<SolanaAddress>&& static_account_keys,
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_transaction.h:62]: What is the reason of changing this? It forces callers to move checked by compiler and has better performance.
Even if the caller uses move, it still does a copy construct. Compare this simple example and the instructions it generates
https://godbolt.org/z/vs1G69947 vs https://godbolt.org/z/K5Mz1hYqc

Comment (supermassive) [reply components/brave_wallet/browser/solana_transaction.h:62]: This guide https://google.github.io/styleguide/cppguide.html#Rvalue_references discourages use of r-value references for arguments.
Godbolt comparisons should use clang with args at least including `-O2 -fno-exceptions` but this immediately over optimizes these examples without obvious difference.

File: components/brave_wallet/browser/tx_storage_delegate_impl.cc [hunk 2]
```diff
@@ -24,24 +25,54 @@ constexpr char kValueStoreDatabaseUMAClientName[] = "BraveWallet";
...
 TxStorageDelegateImpl::TxStorageDelegateImpl(
     PrefService* prefs,
     scoped_refptr<value_store::ValueStoreFactory> store_factory,
-    scoped_refptr<base::SequencedTaskRunner> ui_task_runner) {
-  store_ = std::make_unique<value_store::ValueStoreFrontend>(
-      store_factory, base::FilePath(kWalletStorageName),
-      kValueStoreDatabaseUMAClientName, ui_task_runner,
-      value_store::GetValueStoreTaskRunner());
-  // txs_ has been set during migration.
-  if (!MigrateTransactionsFromPrefsToDB(prefs)) {
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/tx_storage_delegate_impl.cc:40]: Now it is straightforward initialization.
MaybeMigratePrefsToDb
LoadFromDb
MaybeMigrateDbFormat

Omitting LoadFromDb in rare cases of migration would just complicate code and not worth it.

File: components/brave_wallet/browser/tx_state_manager_unittest.cc [hunk 3]
```diff
@@ -123,8 +122,7 @@ TEST_F(TxStateManagerUnitTest, ConvertFromAddress) {
 
   auto txs = GetTxs();
   ASSERT_TRUE(txs);
-  const base::Value::Dict* value =
-      txs->GetDict().FindDictByDottedPath("ethereum.mainnet.001");
+  const base::Value::Dict* value = txs->GetDict().FindDictByDottedPath("001");
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/tx_state_manager_unittest.cc:125]: will fix

**Review by darkdh — APPROVED** [review]
++, please resolve the only one nit before merging.

**Pull request merged**